### PR TITLE
[fins] add libfins

### PR DESCRIPTION
**Pull request by endingly**
If this PR adds a new port, please uncomment and fill out this checklist:

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

And the relevant issue is #32667.

### CODE REVIEW

**Review by MonicaLiu0311 — CHANGES_REQUESTED** [review]

File: ports/fins/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+  "$schema": "https://raw.githubusercontent.com/microsoft/vcpkg-tool/main/docs/vcpkg.schema.json",
+  "name": "fins",
+  "version": "0.7.0",
```
[the remark this answers is not in the document]
Comment (endingly) [reply ports/fins/vcpkg.json:4]: Oh, this version information is found on the official website of this library, not the repository.
This is the website address: https://www.libfins.org/download/

[the remark this answers is not in the document]
Comment (dg0yt) [reply ports/fins/vcpkg.json:4]: I can't find where I can connect a particular version to a particular git ref.

[the remark this answers is not in the document]
Comment (endingly) [reply ports/fins/vcpkg.json:4]: Ok, if we need precise correspondence, then I'll use the date version instead.

**Review by MonicaLiu0311 — APPROVED** [review]

**Comment by MonicaLiu0311** [comment]
All features are tested successfully in the following triplet:
```
x86-windows
x64-windows
x64-windows-static
```
The usage test passed on `x64-windows` (header files found):
```
fins provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(fins CONFIG REQUIRED)
    target_link_libraries(main PRIVATE fins::fins)
```

**Pull request merged**